app/domain/contactable/address_validator.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2012-2020, CVP Schweiz. This file is part of
+#  hitobito_cvp and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_cvp.
+
+module Contactable
+  class AddressValidator
+
+    def validate_people
+      Person.find_each do |person|
+        next unless should_be_validated?(person)
+
+        if invalid?(person.address, person.zip_code, person.town)
+          tag_invalid!(person, person.address)
+        end
+      end
+
+      Person.tagged_with(invalid_tag)
+    end
+
+    private
+
+    def should_be_validated?(person)
+      Settings.addresses.imported_countries.include?(person.country) &&
+        person.tags.exclude?(invalid_override_tag)
+    end
+
+    def invalid?(address, zip_code, town)
```

Comment:
Gibt es einen grund nicht `Address.search` zu verwenden? Ich denke das wäre besser weil konsistenter mit dem typeahead. Zudem bitte auch noch die Hausnummer aus der Adresse parsen und validieren

Reply:
Es gibt aber auch Adressen ohne Hausnummer, und nicht-numerische Hausnummern wie 3a. Ich bin nicht sicher ob das so trivial ist, die da rauszuholen, und wenn es möglich ist bin ich nicht sicher wie man das clever validieren würde.

Reply:
In der Datenbank speichern wir bereits Hausnummern mit Zusatzziffer (zb 3a). Die Volltext Suche kann mit und ohne Hausnummer suchen. Die Klasse `Address::FullTextSearch` kapselt diese Funktionalität. Ich würde vorschlagen, dass wir die auch hier verwenden und die search strategy aus dem Controller dorthin übernehmen.